python/pylibvw.cc

```diff
@@ -283,6 +285,14 @@ vw_ptr my_initialize_with_log(py::list args, py_log_wrapper_ptr py_log)
 
 vw_ptr my_initialize(py::list args) { return my_initialize_with_log(args, nullptr); }
 
+boost::shared_ptr<VW::workspace> merge_workspaces(py::list workspaces)
+{
+  std::vector<const VW::workspace*> const_workspaces;
+  for (size_t i = 0; i < py::len(workspaces); i++)
+  { const_workspaces.push_back(py::extract<VW::workspace*>(workspaces[i])); }
+  return boost::shared_ptr<VW::workspace>(VW::merge_models(const_workspaces).release());
```

Comment:
you should be able to initialize the shared pointer with the unique ptr, should not need to use release?